### PR TITLE
Feature: add dialect and host list provider for RDS MultiAZ DB Cluster

### PR DESCRIPTION
**Pull request by fangli**
### Summary

This PR:

1. add `RdsMysqlClusterDialect`
2. add `RdsMultiAzDbClusterListProvider`
3. refactor `AuroraHostListProvider`
4. test fixes

### By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: wrapper/src/main/java/software/amazon/jdbc/dialect/DialectCodes.java [hunk 1]
```diff
@@ -20,9 +20,11 @@ public class DialectCodes {
   public static final String AURORA_MYSQL = "aurora-mysql";
   public static final String RDS_MYSQL = "rds-mysql";
   public static final String MYSQL = "mysql";
+  public static final String RDS_MULTIAZ_MYSQL_CLUSTER = "rds-maz-mysql-cluster";
```
Comment (sergiyvamz) [inline wrapper/src/main/java/software/amazon/jdbc/dialect/DialectCodes.java:23]: "rds-mysql-taz" for consistency with previous codes?

File: wrapper/src/main/java/software/amazon/jdbc/dialect/RdsMysqlClusterDialect.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+package software.amazon.jdbc.dialect;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.List;
+import software.amazon.jdbc.hostlistprovider.RdsMultiAzDbClusterListProvider;
+
+public class RdsMysqlClusterDialect extends MysqlDialect {
```
Comment (sergiyvamz) [inline wrapper/src/main/java/software/amazon/jdbc/dialect/RdsMysqlClusterDialect.java:26]: It seems that this class is specific to TAZ clusters so I'd recommend to rename it to RdsTazMysqlClusterDialect

File: wrapper/src/main/java/software/amazon/jdbc/dialect/RdsPgClusterDialect.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ * limitations under the License.
+ */
+
+package software.amazon.jdbc.dialect;
+
+/**
+ * Suitable for the following AWS PG configurations.
+ * - Regional Cluster
+ */
+public class RdsPgClusterDialect extends PgDialect {
```
Comment (sergiyvamz) [inline wrapper/src/main/java/software/amazon/jdbc/dialect/RdsPgClusterDialect.java:23]: What's the purpose of the class?

File: wrapper/src/main/java/software/amazon/jdbc/dialect/DialectCodes.java [hunk 4]
```diff
@@ -20,9 +20,11 @@ public class DialectCodes {
   public static final String AURORA_MYSQL = "aurora-mysql";
   public static final String RDS_MYSQL = "rds-mysql";
   public static final String MYSQL = "mysql";
+  public static final String RDS_MULTI_AZ_MYSQL_CLUSTER = "rds-multi-az-mysql-cluster";
```
Comment (sergiyvamz) [inline wrapper/src/main/java/software/amazon/jdbc/dialect/DialectCodes.java:23]: Can we be a more consistent with previous names and remove "_cluster"?

Comment (fangli) [reply wrapper/src/main/java/software/amazon/jdbc/dialect/DialectCodes.java:23]: `Cluster` is part of the product name and we can't remove it.
[MultiAZ DB Cluster](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/multi-az-db-clusters-concepts.html) consists of 3 nodes, while [MultiAZ DB](https://aws.amazon.com/rds/features/multi-az/) consists of 1 active node and 1 standby node.

File: wrapper/src/main/java/software/amazon/jdbc/hostlistprovider/RdsMultiAzDbClusterListProvider.java [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+
+    try {
+      final Statement stmt = conn.createStatement();
+      String writerNodeId = processWriterNodeId(stmt.executeQuery(this.fetchWriterNodeQuery));
+      if (writerNodeId.isEmpty()) {
+        final ResultSet nodeIdResultSet = stmt.executeQuery(this.nodeIdQuery);
+        while (nodeIdResultSet.next()) {
+          writerNodeId = nodeIdResultSet.getString(1);
+        }
+      }
```
Comment (sergiyvamz) [inline wrapper/src/main/java/software/amazon/jdbc/hostlistprovider/RdsMultiAzDbClusterListProvider.java:93]: This block is unclear. If writer is empty, it fetches topology and consider the last node in this topology as a writer. Is that correct?

Comment (fangli) [reply wrapper/src/main/java/software/amazon/jdbc/hostlistprovider/RdsMultiAzDbClusterListProvider.java:93]: This part is still being discussed and subject to change. For current implementation:
* If `SHOW REPLICA STATUS` returns a Source node ID, it implies that the current node is read replica.
* If `SHOW REPLICA STATUS` returns nothing, it means that the current node is not replicating from other nodes, implies that the current node is writer.

**Pull request closed without merging**